apis/v1alpha2/gateway_types.go

```diff
@@ -374,59 +376,33 @@ const (
 	TLSModePassthrough TLSModeType = "Passthrough"
 )
 
-// RouteBindingSelector defines a schema for associating routes with the Gateway.
-// If Namespaces and Selector are defined, only routes matching both selectors are
-// associated with the Gateway.
-type RouteBindingSelector struct {
-	// Namespaces indicates in which namespaces Routes should be selected
-	// for this Gateway. This is restricted to the namespace of this Gateway by
+// ListenerRoutes defines which Routes may be attached to this Listener.
+type ListenerRoutes struct {
+	// Namespaces indicates which namespaces Routes may be attached to this
+	// Listener from. This is restricted to the namespace of this Gateway by
 	// default.
 	//
 	// Support: Core
 	//
 	// +optional
 	// +kubebuilder:default={from: Same}
 	Namespaces *RouteNamespaces `json:"namespaces,omitempty"`
-	// Selector specifies a set of route labels used for selecting
-	// routes to associate with the Gateway. If this Selector is defined,
-	// only routes matching the Selector are associated with the Gateway.
-	// An empty Selector matches all routes.
-	//
-	// Support: Core
-	//
-	// +optional
-	Selector *metav1.LabelSelector `json:"selector,omitempty"`
-	// Group is the group of the route resource to select. Omitting the value
-	// indicates the gateway.networking.k8s.io API group.
-	// For example, use the following to select an HTTPRoute:
-	//
-	// routes:
-	//   kind: HTTPRoute
-	//
-	// Otherwise, if an alternative API group is desired, specify the desired
-	// group:
+
+	// Kinds specifies the groups and kinds of Routes that are allowed to bind
```

Comment:
```suggestion
	// Kinds specifies the groups and kinds of Routes that are allowed to attach
```
for consistency and to save me from wondering whether "bind" and "attach" mean different things... which they don't, right?

Reply:
Not intended to mean anything different, just ended up having more `attach` terminology in the godocs this time around because the field started with an `attachTo` name.